app/controllers/RiderRankingConnectionController.java

```diff
@@ -0,0 +1,116 @@
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Executors;
+
+import static play.libs.Json.toJson;
+
+public class RiderRankingConnectionController extends Controller {
+    private final RiderRankingRepository riderRankingRepository;
+
+    @Inject
+    public RiderRankingConnectionController(RiderRankingRepository riderRankingRepository) {
+        this.riderRankingRepository = riderRankingRepository;
+    }
+
+
+    public CompletionStage<Result> getRiderRankings(long riderStageConnectionId) {
+        return riderRankingRepository.getAllRiderRankings(riderStageConnectionId).thenApplyAsync(riderRankings -> {
+            return ok(toJson(riderRankings));
+        }).exceptionally(ex -> {
+            Result res = null;
+            switch (ExceptionUtils.getRootCause(ex).getClass().getSimpleName()){
+                case "IndexOutOfBoundsException":
+                    res = badRequest("No rider rankings are set in DB.");
+                    break;
+                default:
+                    res = internalServerError(ex.getMessage());
+            }
+            return res;
+        });
+    }
+
+    public CompletionStage<Result> getRiderRankingsByType(long riderStageConnectionId, String rankingType) {
+        return riderRankingRepository.getAllRiderRankingsByType(riderStageConnectionId, rankingType).thenApplyAsync(riderRankings -> {
+            return ok(toJson(riderRankings));
+        }).exceptionally(ex -> {
+            Result res = null;
+            switch (ExceptionUtils.getRootCause(ex).getClass().getSimpleName()){
+                case "IndexOutOfBoundsException":
+                    res = badRequest("No rider rankings are set in DB.");
+                    break;
+                default:
+                    res = internalServerError(ex.getMessage());
+            }
+            return res;
+        });
+    }
+
+    public CompletionStage<Result> getRiderRankingByRiderAndType(long riderId, String rankingType) {
+        return riderRankingRepository.getRiderRankingByRiderAndType(riderId, rankingType).thenApplyAsync(riderRanking -> {
+            return ok(toJson(riderRanking));
+        }).exceptionally(ex -> {
+            Result res = null;
+            switch (ExceptionUtils.getRootCause(ex).getClass().getSimpleName()){
+                case "IndexOutOfBoundsException":
+                    res = badRequest("No rider ranking is set in DB.");
+                    break;
+                default:
+                    res = internalServerError(ex.getMessage());
+            }
+            return res;
+        });
+    }
+
+    @BodyParser.Of(BodyParser.Json.class)
+    public CompletionStage<Result> updateRiderRanking(long riderRankingId) {
+        JsonNode json = request().body().asJson();
+        return parseRiderRanking(json, riderRankingId).thenApply(riderRankingRepository::updateRiderRanking).thenApplyAsync(rSC -> {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove useless curly braces around statement and then remove useless return keyword (sonar.java.source not set. Assuming 8 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1602)